e2e-tests/common.sh

```diff
@@ -51,26 +51,99 @@ function setup() {
   http://localhost:8099/openmrs/ws/fhir2/R4/Encounter/$2 2>/dev/null >>./omrs/encounters.json
-  total_test_encounters=$(jq '.total' ./omrs/encounters.json)
+  local total_test_encounters=$(jq '.total' ./omrs/encounters.json)
   print_message "Total openmrs test encounters ---> ${total_test_encounters}"
   curl -L -X GET -u admin:Admin123 --connect-timeout 5 --max-time 20 \
   http://localhost:8099/openmrs/ws/fhir2/R4/Observation/$3 2>/dev/null >>./omrs/obs.json
-  total_test_obs=$(jq '.total' ./omrs/obs.json)
+  local total_test_obs=$(jq '.total' ./omrs/obs.json)
   print_message "Total openmrs test obs ---> ${total_test_obs}"
   print_message "Counting number of patients, encounters and obs sinked to parquet files"
   total_patients_streamed=$(java -jar ./parquet-tools-1.11.1.jar rowcount ./Patient/ | awk '{print $3}')
   print_message "Total patients synced to parquet ---> ${total_patients_streamed}"
   total_encounters_streamed=$(java -jar ./parquet-tools-1.11.1.jar rowcount ./Encounter/ | awk '{print $3}')
   print_message "Total encounters synced to parquet ---> ${total_encounters_streamed}"
   total_obs_streamed=$(java -jar ./parquet-tools-1.11.1.jar rowcount ./Observation/ | awk '{print $3}')
-  print_message "Total obs synced to parquet ---> ${total_obs_streamed}"
+  print_message "Total obs synced to parquet ---> ${total_obs_streamed}"  
+  if [[ -z "$1" ]]; then
+    if [[ ${total_patients_streamed} == ${total_test_patients} && ${total_encounters_streamed} == ${total_test_encounters} && ${total_obs_streamed} == ${total_test_obs} ]] ; then
+        print_message "BATCH MODE WITH PARQUET SYNC EXECUTED SUCCESSFULLY USING ${mode} MODE"
+        cd "${HOME_PATH}"
+    else    
+        print_message "BATCH MODE WITH PARQUET SYNC TEST FAILED USING ${mode} MODE"
+        return 1
+    fi 
+  else 
+    if [[ ${total_patients_streamed} == ${total_test_patients} && ${total_encounters_streamed} == ${total_test_encounters} && ${total_obs_streamed} == ${total_test_obs} ]] ; then
+        print_message "STREAMING MODE WITH PARQUET SINK EXECUTED SUCCESSFULLY"
+    else    
+        print_message "STREAMING MODE WITH PARQUET SINK TEST FAILED"
+        return 1
+    fi
+   return 1 
+  fi        
+}
+
+###################################################################################
+# Function that validates fhir counts for batch and stream mode
+# Arguments:
+#   ?family=Clive = Argument # 1 passed to the function (positional parameter # 1)
+###################################################################################
+function validate_fhir_counts() {
+  mkdir batch_omrs
+  mkdir stream_fhir
+  mkdir batch_fhir
+  mkdir stream_omrs
+  if [[ -z "$1" ]]; then
+    print_message "Finding number of patients, encounters and obs in openmrs server"
+    curl -L -X GET -u admin:Admin123 --connect-timeout 5 --max-time 20 \
+    http://localhost:8099/openmrs/ws/fhir2/R4/Patient/ 2>/dev/null >>./batch_omrs/patients.json
```

Comment:
nit: Please indent (2 extra spaces) when breaking a line with `\` (here and elsewhere).